pkg/networkservice/common/connect/server.go

```diff
@@ -26,29 +26,40 @@ import (
 	"github.com/golang/protobuf/ptypes/empty"
 	"github.com/pkg/errors"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
 
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
 
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/clienturl"
-	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/inject/injecterror"
-	"github.com/networkservicemesh/sdk/pkg/tools/clientmap"
 	"github.com/networkservicemesh/sdk/pkg/tools/clienturlctx"
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcutils"
 	"github.com/networkservicemesh/sdk/pkg/tools/log"
+	"github.com/networkservicemesh/sdk/pkg/tools/multiexecutor"
 )
 
+type closeFunc func()
+
 type connectServer struct {
 	ctx               context.Context
 	clientFactory     func(ctx context.Context, cc grpc.ClientConnInterface) networkservice.NetworkServiceClient
 	clientDialOptions []grpc.DialOption
-	connInfos         connectionInfoMap
-	clients           clientmap.RefcountMap
+
+	connInfos connectionInfoMap
+	clients   clientInfoMap
+	executor  multiexecutor.MultiExecutor
+}
+
+type clientInfo struct {
+	client  networkservice.NetworkServiceClient
+	count   int
+	onClose closeFunc
```

Comment:
Looks like `onClose` is a `context.CancelFunc` wrapped with `closeFunc`. Do we need this additional type?

Reply:
And so it can be `cancel`, not `onClose`.

Reply:
Done